### PR TITLE
Feature/qw003 add question set

### PR DESCRIPTION
**Pull request by Misvor**

### CODE REVIEW

**Review by Sar-dim — APPROVED** [review]
All is okay in my opinion

File: QuizWhois.Api/Controllers/QuestionController.cs [hunk 1]
```diff
@@ -24,5 +26,14 @@ public IActionResult Post(QuestionModel questionModel)
             var addedOperation = _questionService.AddQuestion(questionModel);
             return Ok(addedOperation);
         }
+        
+        [HttpPost("AddMany")]
```
Comment (sidgrouse) [inline QuizWhois.Api/Controllers/QuestionController.cs:30]: lets use consistent naming notation and avoid using verbs

File: QuizWhois.Api/Controllers/QuizController.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+using QuizWhois.Common.Models;
+using QuizWhois.Domain.Services.Interfaces;
+
+namespace QuizWhois.Api.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class QuizController
+    {
+
```
Comment (sidgrouse) [inline QuizWhois.Api/Controllers/QuizController.cs:16]: seems like code rules are not applied here for some reason

File: QuizWhois.Api/Controllers/QuizController.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        private readonly IQuizService _quizService;
+
+        public QuizController(IQuizService quizService)
+        {
+            _quizService = quizService;           
+        }
+
+        // [Authorize]
+        [HttpPost("AddToSet")]
```
Comment (sidgrouse) [inline QuizWhois.Api/Controllers/QuizController.cs:25]: why not to use something like HttpPost("{quizId}")?

File: QuizWhois.Api/tests.analyzers.ruleset [hunk 4]
```diff
@@ -0,0 +1,108 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (sidgrouse) [inline QuizWhois.Api/tests.analyzers.ruleset:1]: Is not it better to move the file into the solution root and just reference from each project? So we can keep it consistent and avoid copy-pasting

File: QuizWhois.Domain/Services/Interfaces/IQuizService.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using QuizWhois.Common.Models;
+
+
+namespace QuizWhois.Domain.Services.Interfaces
+{
+    public interface IQuizService
+    {
+        public Task<QuizModel> AddToQuiz(long quiz, long question); //, int index
```
Comment (sidgrouse) [inline QuizWhois.Domain/Services/Interfaces/IQuizService.cs:13]: what is the matter with int index?

File: QuizWhois.Domain/Services/Implementations/QuizService.cs [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+            quizToSave.Questions = questionsOfQuiz.ToList();
+            _db.Set<Quiz>().Add(quizToSave);
+            await _db.SaveChangesAsync();
+            return new QuizModel(quizToSave.Id, quizToSave.Questions.Select(x => x.Id));
+        }
+
+        public async Task<QuizModel> AddToQuiz(long quizId, Question question)
+        {
+            var quiz = await _db.Quizzes.FirstOrDefaultAsync(x => x.Id == quizId);
+            if (quiz == null)
```
Comment (sidgrouse) [inline QuizWhois.Domain/Services/Implementations/QuizService.cs:35]: lets discuss throwing an exception on such an occasion

**Review by sidgrouse — CHANGES_REQUESTED** [review]

File: QuizWhois.Api/Controllers/QuizController.cs (hunk 3 above)
Comment (Misvor) [reply QuizWhois.Api/Controllers/QuizController.cs:25]: But i would consider [HttpPost("AddToSet/{quizId}")]

Comment (Misvor) [reply QuizWhois.Api/Controllers/QuizController.cs]: @sidgrouse Something like that? 
image

**Review by sidgrouse — APPROVED** [review]

**Pull request merged**